Syntax types

akanellop opened the pull request:
separation of values from the env variables 
validation checking of types' value
implementation of tests for the above

akanellop committed:
first parsing between values and variables

akanellop committed:
first parsing between values and variables

akanellop committed:
first parsing between values and variables

parisk commented on src/main.test.js:
```diff
@@ -27,3 +27,18 @@ test("Valid input", () => {
 });
+
+test("Wrong value type", () => {
+  const testEnv4 = "DATABASE_URL: notgood\nADMIN_EMAIL\nDEBUG :   text";
```
This should result in an error. Even a single invalid line in a `.env.spec` file should render nothing and return `null`.

parisk commented on src/main.js:
```diff
@@ -1,29 +1,51 @@
+  let parsPos = -1;
+  return envArrayParsed = envSpecAsArray.map( element =>{
+    parsPos = element.indexOf(":");
+    if(parsPos<0){
```
We should simplify this. We can use the [`includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) method, present on strings.

parisk commented on src/main.js:
```diff
@@ -1,29 +1,51 @@
 //function that checks if the env value is valid
 //value cannot start with a digit and must only contain alphanumeric characters or an underscore
 const checkValidationOfValues = envSpecString => {
+  const validValues = ['color','date','datetime-local','email','month','number','password','tel','text','time','url','week','default'];
```
Two things here:

1. We should rename this to `validTypes`
2. We should remove `default` from the valid types. There is no `default` type in the provided link in https://github.com/sourcelair/env-spec-js/issues/3

parisk commented on src/main.js:
```diff
@@ -1,29 +1,51 @@
+  })
 };
 
+//function that separates the variables from their values and returns them as separated arrays
```
> separates the variables from their values

I guess the appropriate wording here is; _separates the variables from their **types**_

parisk commented on src/main.js:
```diff
@@ -1,29 +1,51 @@
 };
 
+//function that separates the variables from their values and returns them as separated arrays
+const parseVarFromVal = envSpecAsArray => {
```
Can you describe in a comment the format of expected input of this function and the format of its output please?

parisk commented on src/main.js:
```diff
@@ -1,29 +1,51 @@
+    parsPos = element.indexOf(":");
+    if(parsPos<0){
+      parsPos=element.length
+      return [element.substring(0,parsPos).trim(),'default'];
```
There is no `default` type. We should default to `text` instead.

parisk commented on src/main.js:
```diff
@@ -1,29 +1,51 @@
-      `<label for="env_spec_${element.toLowerCase()}">${element}</label>\n` +
-      `<input id="env_spec_${element.toLowerCase()}" name="${element.toLowerCase()}" />\n`
+      `<label for="env_spec_${element[0].toLowerCase()}">${element[0]}</label>\n` +
+      `<input id="env_spec_${element[0].toLowerCase()}" name="${element[0].toLowerCase()}" />\n`
```
We should render the `input`'s type here as described in the example here: https://github.com/sourcelair/env-spec-js/issues/3.

parisk commented on src/main.js:
```diff
@@ -1,29 +1,51 @@
   return outputHTML(checkValidationOfValues(envSpec));
 };
 
+console.log(envSpecToHTML("DATABASE_URL: notgood\nADMIN_EMAIL\nDEBUG :   text"));
```
Let's remove this from here. We do not want our libraries printing things in users' consoles 😄.

parisk reviewed (changes requested):
Alright! Did a few comments here.

**A couple more**:

1. we should add a test case for valid input of "typed" environment variables
2. we should also test mixed input of both typed and untyped environment variables

**P.S.:** Untyped environment variables should default to the `text` input type.

akanellop committed:
- changed parseVarFromType function using includes and split methods
- changed comments
- changed tests
-
- envSpecToHTML method can parse and validate enviromental variables anf their types

akanellop committed:
Fix typo

parisk reviewed (changes requested):
Almost there! Before we move on with merging this, we need a couple of things done:

1. Change the commit message of your last commit (https://github.com/sourcelair/env-spec-js/pull/6/commits/faa4704eeab88bd9af6a85e809ab062a6b0f22e5), which has the same commit message with the previous one (https://github.com/sourcelair/env-spec-js/pull/6/commits/76676c3b5483b5d1f6ba58ae4a7a9f1c91bd9f02) to something more semantic (`Fix typo` looks like an appropriate message). You can do this witj `git commit --amend`.
2. Run the `format` yarn script to format your files with prettier

I think we are good to go after these two changes.

akanellop committed:
Files formatted with "prettier" method

parisk reviewed (approved):
Amazing 🎊! Thanks 🙏.
